File: src/main/scala/edu/osu/cse/groenkeb/logic/Atom.java
package edu.osu.cse.groenkeb.logic;

public final class Atom
{
  public static final Atom ABSURDITY = new Atom("!");
  
  private final String name;

  public Atom(String name)
  {
    this.name = name;
  }

  public boolean matches(Atom atom)
  {
    assert name != null;
    return this.name.equals(atom.name);
  }

  @Override
  public String toString()
  {
    return name;
  }
}

File: src/main/scala/edu/osu/cse/groenkeb/logic/Atom.scala
package edu.osu.cse.groenkeb.logic

case class Atom(val name: String) {
  def matches(atom: Atom) = atom != null && name.equals(atom.name)
  
  override def toString = name
}

object Atom {
  final val absurdity = new Atom("!")
}